sysversion/__init__.py

```diff
@@ -9,38 +9,66 @@
 # License, or (at your option) any later version.
 
 import re
-from subprocess import run, PIPE
-from os.path import join
+import subprocess
+from subprocess import PIPE
+from pathlib import Path
 from sys import stdin
 
 
+DEFAULT = 'etc/turnkey_version'
+
+
 def _parse_turnkey_release(version):
     m = re.match(r'turnkey-.*?-(\d.*?)-[^\d]', version)
     if m:
         return m.group(1)
 
 
+def get_debian_codename(encoding=stdin.encoding):
+    """Return Debian codename of the system (leverages lsb_release)"""
+    process = subprocess.run(['lsb_release', '-sc'], stdout=PIPE)
+    if process.returncode != 0:
+        return
+    return process.stdout.decode(encoding).strip()
```

Comment:
I believe `subprocess.run(...)` takes arguments `text: bool` and `encoding: str` which can be used to automatically decode stdin, stdout & stderr. This *might* be cleaner

Reply:
Yep, [`subprocess.run()`](https://docs.python.org/3/library/subprocess.html#subprocess.run) does take `text=` &/or `encoding=` args. I was aware of `encoding=`, but TBH I didn't realise that was the use case of `text=`... I'll have a bit of a play and see how we go...